File: graphql-subscriptions-rails/app/graphql/input_types/post_input_type.rb
# frozen_string_literal: true

InputTypes::PostInputType = GraphQL::InputObjectType.define do
  name 'PostInputType'
  description 'Properties for creating a Post'

  input_field :title, !types.String, 'Title of the post.'
  input_field :body, !types.String, 'Body of the post.'
end

File: graphql-subscriptions-express/graphql/post_input_type.js
const {GraphQLInputObjectType, GraphQLString, GraphQLNonNull} = require('graphql');


const PostInputType = new GraphQLInputObjectType({
    name: 'PostInput',
    fields: {
        title: { type: new GraphQLNonNull(GraphQLString) },
        body: { type: new GraphQLNonNull(GraphQLString) },
    }
});

module.exports = PostInputType;
